eth/config.go

```diff
@@ -149,4 +150,10 @@ type Config struct {
 
 	// RPCGasCap is the global gas cap for eth-call variants.
 	RPCGasCap *big.Int `toml:",omitempty"`
+
+	// Checkpoint is a hardcoded checkpoint which can be nil.
+	Checkpoint *params.TrustedCheckpoint
+
+	// CheckpointConfig is a set of checkpoint contract configs.
+	CheckpointConfig *params.CheckpointContractConfig
```

Comment:
Let's call this `CheckpointOracle` and `params.CheckpointOracleConfig`. Since we moved it out of the ChainConfig, Oracle seems a bit more appropriate name here.